sale_quotation_products/models/sale_order.py

```diff
@@ -34,8 +34,8 @@ def add_products_to_quotation(self):
     @api.multi
```

Comment:
1. > from ast import literal_eval

use odoo save eval instead `from odoo.tools.safe_eval import safe_eval`

2. > instead of update context one by one, use only one update()
```python
   context.update(dict(
        sale_quotation_products=True,
        pricelist=self.pricelist_id.display_name,
        # we send company in context so it filters taxes
        company_id=self.company_id.id,
        partner_id=self.partner_id.id,
        search_default_location_id=self.warehouse_id.lot_stock_id.id,
         # search_default_warehouse_id=self.warehouse_id.id,
    )
```
3. > same for action_read

```python
    action_read.update(
        context=context,
        # view_mode='tree,form'.
        name=_('Quotation Products'),
    )
```